cmd/prow-job-dispatcher/main.go

```diff
@@ -370,4 +430,67 @@ func main() {
 	if err := dispatcher.SaveConfig(config, o.configPath); err != nil {
 		logrus.WithError(err).Fatalf("Failed to save config file to %s", o.configPath)
 	}
+
+	if !o.createPR {
+		logrus.Info("Finished dispatching and create no PR, exiting ...")
+		os.Exit(0)
+	}
+
+	logrus.WithField("targetDir", o.targetDir).Info("Changing working directory ...")
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+
+	steps := []struct {
+		command   string
+		arguments []string
+	}{
+		{
+			// start with git commands without running any tools
+			command: "",
```

Comment:
This is still super confusing, removing the `run` part out of the `runAndCommitIfNeeded` function and not trying to use a loop here would made everything more obvious

Reply:
That was one of my commits and I felt using the loop is simpler.
I will follow up tomorrow.